functions/data_transformations.R

```diff
@@ -0,0 +1,77 @@
+library(dplyr)
+library(purrr)
+library(tidyr)
+library(lubridate)
+library(jsonlite)
+
+# Function to transfrm metadata to a readable dataframe
+transform_metadata_to_df <- function(metadata) {
+  transformed_data <- metadata[[1]] %>%
+    map(as_tibble) %>%
+    list_rbind() %>%
+    mutate(
+      latestData = map_chr(
+        latestData,
+        1,
+        .default = NA_character_
+      )) %>%
+    mutate(latestData = as_datetime(latestData, tz = "UTC")) %>%
+    unnest_wider(location) %>%
+    unnest_wider(latLon)
+  
+  return(transformed_data)
+}
+
+
+#Function to return a date with the offset added
+
+to_iso8601 <- function(datetime, offset_days) {
+  #Converting the inpu to a date-time object
+  
+  datetime_obj <- as_datetime(datetime)
+  #Applying the offset
+  
+  datetime_obj <- datetime_obj + days(offset_days)
+  
+  #Formating the date-time as an ISO8601 string and append "Z" 
+  iso8601_str <- format(datetime_obj, "%Y-%m-%dT%H:%M:%S") %>% 
+    paste0("Z")
+  
+  return(iso8601_str)
+}
+
+#Testing to_iso8601 function
+source('functions/data_transformations.r')
```

Comment:
This causes an error in the iterations.r script, because you specify the path 'functions/data_transformations.r' inside the data_transformation.r script. 

You are using the source() function on the same script you are in, which doesnt make sense.